milkman-sio/src/main/java/milkman/plugin/sio/editor/SocketIOAspectEditor.java

```diff
@@ -0,0 +1,132 @@
+package milkman.plugin.sio.editor;
+
+import com.jfoenix.controls.JFXButton;
+import com.jfoenix.controls.JFXTextField;
+
+import de.jensd.fx.glyphs.fontawesome.FontAwesomeIcon;
+import de.jensd.fx.glyphs.fontawesome.FontAwesomeIconView;
+import javafx.scene.control.ContentDisplay;
+import javafx.scene.control.Label;
+import javafx.scene.control.Tab;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Priority;
+import javafx.scene.layout.VBox;
+import milkman.ctrl.ExecutionListenerManager;
+import milkman.domain.RequestContainer;
+import milkman.domain.ResponseContainer;
+import milkman.plugin.sio.domain.SocketIOAspect;
+import milkman.plugin.sio.domain.SocketIOResponseAspect;
+import milkman.ui.components.ContentEditor;
+import milkman.ui.contenttype.PlainContentTypePlugin;
+import milkman.ui.plugin.ExecutionListenerAware;
+import milkman.ui.plugin.RequestAspectEditor;
+import milkman.ui.plugin.rest.contenttype.JsonContentType;
+import milkman.ui.plugin.rest.domain.RestHeaderAspect;
+import milkman.utils.fxml.GenericBinding;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+import static milkman.utils.FunctionalUtils.run;
+
+public class SocketIOAspectEditor implements RequestAspectEditor, ExecutionListenerAware {
+
+    private ExecutionListenerManager executionListenerManager;
+
+
+    @Override
+    public Tab getRoot(RequestContainer request) {
+        return getRoot(request, Optional.empty());
+    }
+
+    @Override
+    public Tab getRoot(RequestContainer request, Optional<ResponseContainer> existingResponse) {
+        SocketIOAspect sioAspect = request.getAspect(SocketIOAspect.class).get();
+
+        TextField textField = new JFXTextField();
+		GenericBinding<SocketIOAspect, String> binding = GenericBinding.of(
+            SocketIOAspect::getEvent, 
+            run(SocketIOAspect::setEvent)
+                .andThen(() -> sioAspect.setDirty(true)), //mark aspect as dirty propagates to the request itself and shows up in UI
+            sioAspect); 
+		textField.textProperty().bindBidirectional(binding);
+		textField.setUserData(binding); //need to add a strong reference to keep the binding from being GC-collected.
+		HBox.setHgrow(textField, Priority.ALWAYS);
+
+        ContentEditor editor = new ContentEditor();
```

Comment:
might make sense to just hide the header as this obfuscates the UI.  (`editor.setHeaderVisibility(false)`)

Reply:
I thought it would be nice to give option to the user to switch message type between plain text and json, like in rest request